### PR TITLE
Add 'All' targeting strategy

### PR DESCRIPTION
**Pull request by tkz00**
Closes [Champions of Mirra - Targeting Strategy All](https://github.com/lambdaclass/champions_of_mirra/issues/405)

Added new strategy to target all units in one team or another, not all units in the battle.

## Summary of changes

New `choose_targets` function clause with the 'all' type of targeting strategy, it returns all the units on the caster unit's ally team (including the caster unit) or opposite team, depending of the `target_allies` parameter.

## How to test it?

Set the `targeting strategy` for one of your unit character's skills to "all", you can do so in the `skills.json` file, running `Champions.Config.import_skill_config()` to update the running backend. Then run a battle and verify when the modified skill is cast, it targeted the whole team it was meant for. You can check battle.md on how to run battles.

## Checklist
- [X] Tested the changes locally.
- [X] Reviewed the changes on GitHub, line by line.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Comment by lotuuu** [comment]
This looks good! How do you feel about writing a test for this strategy? You can check the other tests in "Targeting Strategies" section in `battle_test.ex` for examples. This one could go like: a unit with "all" targeting strategy that deals 10 damage vs 6 target dummies with 10hp, that ends in team 1 victory after only 1 skill cast

**Comment by tkz00** [comment]
> xThis looks good! How do you feel about writing a test for this strategy? You can check the other tests in "Targeting Strategies" section in `battle_test.ex` for examples. This one could go like: a unit with "all" targeting strategy that deals 10 damage vs 6 target dummies with 10hp, that ends in team 1 victory after only 1 skill cast.
> 
> If you're not comfortable with tests I can give you a hand with it but I think you'll be fine!

Yeah! I wasn't sure about which way to test this, but that sounds nice, I'll add it now.

File: apps/champions/test/battle_test.exs [hunk 1]
```diff
@@ -679,4 +679,69 @@ defmodule Champions.Test.BattleTest do
...
+        base_attack: 10,
+        base_health: 10
+      })
+      |> Characters.insert_character()
+
+    {:ok, unit} = TestUtils.build_unit(%{character_id: character.id}) |> Units.insert_unit()
+    {:ok, unit} = Units.get_unit(unit.id)
+
+    # Create 6 target dummies for enemy team
+    [target_dummy_1, target_dummy_2, target_dummy_3, target_dummy_4, target_dummy_5, target_dummy_6] =
```
Comment (lotuuu) [inline apps/champions/test/battle_test.exs:730]: Let's just capture this whole list as `target_dummies`

File: apps/champions/test/battle_test.exs [hunk 2]
```diff
@@ -679,4 +679,69 @@ defmodule Champions.Test.BattleTest do
...
+                }
+              })
+          }
+        ],
+        cooldown: maximum_steps * @miliseconds_per_step - 1
+      })
+
+    {:ok, character} =
+      TestUtils.build_character(%{
+        name: "Character",
```
Comment (lotuuu) [inline apps/champions/test/battle_test.exs:718]: Name this character with a more specific name, so as to not get any "name taken" errors. Same with the "Empty Skill". You could name them "All Character" and "All Empty Skill".

**Review by lotuuu — CHANGES_REQUESTED** [review]
Looking great! Some small comments

File: apps/champions/test/battle_test.exs (hunk 1 above)
Comment (tkz00) [reply apps/champions/test/battle_test.exs]: Fixed!

File: apps/champions/test/battle_test.exs (hunk 2 above)
Comment (tkz00) [reply apps/champions/test/battle_test.exs]: Also fixed!

**Review by lotuuu — DISMISSED** [review]

**Review by agustinesco — APPROVED** [review]
LGTM!

**Review by ncontinanza — APPROVED** [review]
LGTM! Tested using the client, everything worked great

**Pull request merged**